test/hotspot/jtreg/runtime/os/TestTracePageSizes.java

```diff
@@ -33,59 +35,75 @@
  * @test id=explicit-large-page-size
  * @summary Run test explicitly with both 2m and 1g pages on x64. Excluding ZGC since
  *          it fail initialization if no large pages are available on the system.
+ * @library /test/lib
+ * @build jdk.test.lib.Platform
  * @requires os.family == "linux"
  * @requires os.arch=="amd64" | os.arch=="x86_64"
  * @requires vm.gc != "Z"
- * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseLargePages -XX:LargePageSizeInBytes=2m TestTracePageSizes
- * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseLargePages -XX:LargePageSizeInBytes=1g TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xmx128m -Xlog:pagesize:ps-%p.log -XX:+UseLargePages -XX:LargePageSizeInBytes=2m TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xmx128m -Xlog:pagesize:ps-%p.log -XX:+UseLargePages -XX:LargePageSizeInBytes=1g TestTracePageSizes
```

Comment:
Just realized that this test should probably not set `-Xmx128m` since we try to use 1g pages. What will happen is that the heap will be rounded up to 1G (the alignment logic here is a bit strange) so we will still test it, but it feels a bit backwards. Maybe set the heap to 1g in this case?

Reply:
Good point.